src/parentheses/Solution.java

```diff
@@ -0,0 +1,34 @@
+package parentheses;
+
+import java.util.Stack;
+
+public class Solution {
+    public boolean isValid(String parentheses) {
+
+        char[] arrayParentheses = parentheses.toCharArray();            // O(parentheses)
+                                                                        // time          Space
+        Stack<Character> stack = new Stack<>(); // O(1)
+        boolean result = true;
+        for (int i = 0; i < arrayParentheses.length; i++) {            // O(arrayParentheses.length);  O(1)
+            char character = arrayParentheses[i];                      // O(1);         O(1)
+
+            if (character == '[' ||                                    // O(1)         O(0)
+                    character == '{' ||
+                    character == '(') {
```

Comment:
This is correct, however, it would be easier to read if we do it like this:
```java
boolean isOpenCharacter = character == '[' || character == '{' || character == '('
if (isOpenCharacter) {
```